AdobeStockAsset/Model/AssetRepository.php

```diff
@@ -121,28 +177,34 @@ public function delete(AssetInterface $asset): void
      */
     public function getList(SearchCriteriaInterface $searchCriteria): AssetSearchResultsInterface
     {
-        /** @var AssetCollection $collection */
-        $collection = $this->collectionFactory->create();
-        $this->joinProcessor->process(
-            $collection,
-            AssetInterface::class
-        );
-
-        $this->collectionProcessor->process($searchCriteria, $collection);
-
-        $items = [];
-        /** @var AssetInterface $item */
-        foreach ($collection->getItems() as $item) {
-            $item->setKeywords($this->assetKeywordRepository->getAssetKeywords($item));
-            $items[] = $item;
+        try {
+            /** @var AssetCollection $collection */
+            $collection = $this->collectionFactory->create();
+            $this->joinProcessor->process(
+                $collection,
+                AssetInterface::class
+            );
+
+            $this->collectionProcessor->process($searchCriteria, $collection);
+
+            $items = [];
+            /** @var AssetInterface $item */
+            foreach ($collection->getItems() as $item) {
+                $item->setKeywords($this->getAssetKeywords->execute($item->getId()));
```

Comment:
There is no need to include properties of Media Gallery Asset with Adobe Stock Asset.
For example to get keywords from asset: `$mediaGalleryGetById->execute($asset->getMediaGalleryId())->getKeywords()`

Reply:
Keywords are not Adobe Stock asset's property and should be set to it